### DISCUSSION TITLE
UxPlay v1.52

### DISCUSSION THREAD

**Discussion by fduncanh** [FDH2/UxPlay#95] [Announcements]
UxPlay v1.51 Latest

UxPlay is a GPLv3 unix AirPlay server: its main use is to act like an AppleTV for screen-mirroring (with audio) of iOS/MacOS clients (iPads, iPhones, MacBooks, etc.) in a window on the server display (with the possibility of sharing that window on screen-sharing applications such as Zoom) on a host running Linux, macOS, or other unix, using Apple's AirPlay Mirror protocol. Older iOS clients v. 9.3.5 and later are supported. When the client screen is not mirrored, Apple Lossless (ALAC) audio can be sent from the client to the server using Airplay protocol. Although UxPlay was originally targeted at x86/x86_64 Server and Desktop systems, it now also supports both 32- and 64-bit Raspberry Pi Arm platforms with hardware-accelerated video decoding (using Video4Linux2, the replacement for deprecated 32-bit-only OpenMAX on the Pi).

This work is based on https://github.com/FD-/RPiPlay, with GStreamer integration from https://github.com/antimof/UxPlay. Tested on Debian 10.11 "Buster" and 11.2 "Bullseye", Ubuntu 20.04 and 21.10, Linux Mint 20.3, Pop!_OS 21.10 (NVIDIA edition), Raspberry Pi OS (bullseye), Rocky Linux 8.5 (a CentOS successor), OpenSUSE 15.3, Manjaro 20.2, macOS 12.3 (Intel and M1), FreeBSD 13.0.

1.52 2022-05-05 Cleaned up initial audio sync code, and reformatted streaming debug output (readable aligned timestamps with decimal points in seconds). Eliminate memory leaks (found by valgrind).

<hr /><em>This discussion was created from the release <a href='https://github.com/FDH2/UxPlay/releases/tag/v1.52'>UxPlay v1.52</a>.</em>

**Comment by suyash-srivastava-dev** [comment]
Hi Team,
I am not sure if this is the correct place to ask but below is my query.

It was working great with Ubuntu 20.04 , but as I updated to 22.04 . I am now getting below errors for GStreamer. Can anyone suggest on this ?

(uxplay:44148): GStreamer-CRITICAL **: {{numbers}}: gst_caps_from_string: assertion 'string' failed

(uxplay:44148): GStreamer-CRITICAL **: {{numbers}}: gst_pad_template_new: assertion 'caps != NULL' failed

(uxplay:44148): GStreamer-CRITICAL **: {{numbers}}: gst_mini_object_unref: assertion 'mini_object != NULL' failed

(uxplay:44148): GStreamer-CRITICAL **: {{numbers}}: gst_element_class_add_pad_template: assertion 'GST_IS_PAD_TEMPLATE (templ)' failed

(uxplay:44148): GStreamer-Video-CRITICAL **: {{numbers}}: gst_video_decoder_init: assertion 'pad_template != NULL' failed

(uxplay:44148): GStreamer-WARNING **: {{numbers}}: Element vaapidecode0 has an ALWAYS template src, but no pad of the same name
